catchup/catchpointService.go

```diff
@@ -523,6 +547,15 @@ func (cs *CatchpointCatchupService) processStageSwitch() (err error) {
 	return nil
 }
 
+// stopOrAbort is called when any of the stage processing function sees that cs.ctx has been canceled. It can be
```

Comment:
I find the following confusing: calling stopOrAbort will cancel cs.ctx which is already canceled. 

Let me describe how I understand it: 
- cs.abort does two things: 
       (A) fix internal state: ResetStagingBalances and updateNodeCatchupMode
       (B) cancel cs.ctx
- stopOrAbort will be called if cs.ctx has been canceled. Then, it calls cs.abort

In effect, stopOrAbort will, among other things (the (A) part above), will cancel cs.ctx which is already canceled. 

Will make it clearer, if the (A) part is removed from abort into a separate func, and called from abort and stopOrAbort.

Reply:
I think that the part that there is one point that you didn't mentioned here : 
The `abort` function is doing the following:
```
	cs.updateNodeCatchupMode(false)
	// we want to abort the catchpoint catchup process, and the node already reverted to normal operation.
	// at this point, all we need to do is to abort the run function.
	cs.cancelCtxFunc()
```
note that the second portion is canceling the context created in `updateNodeCatchupMode`. This is required, since it's a "brand new context".

The `abort` is called due to either external `Abort` call, or due to internal problem.

Reply:
> note that the second portion is canceling the context created in `updateNodeCatchupMode`. This is required, since it's a "brand new context".

Okay, this makes sense now. 
Can you please add a comment about this when cs.cancelCtxFunc() is called in abort?

Reply:
done.